onnx/helper.py

```diff
@@ -506,6 +533,30 @@ def make_sequence_value_info(
     return value_info_proto
 
 
+def make_optional_value_info(
+        name,  # type: Text
+        elem_type,  # type: int
+        shape,  # type: Optional[Sequence[Union[Text, int]]]
+        doc_string="",  # type: Text
+        elem_shape_denotation=None,  # type: Optional[List[Text]]
+):  # type: (...) -> ValueInfoProto
+    """Makes a ValueInfoProto based on the data type and shape for Optional."""
+    value_info_proto = ValueInfoProto()
+    value_info_proto.name = name
+    if doc_string:
+        value_info_proto.doc_string = doc_string
+
+    optional_type_proto = value_info_proto.type.optional_type
+    optional_type_proto.elem_type.tensor_type.elem_type = elem_type
```

Comment:
make_optional_value_info will only work for tensors and not sequences or sparse tensors?

Reply:
This was replicating the behavior of make_sequence_value_info. I extended this for to support sequences as well.

Reply:
Is "elem_type" supposed to be OptionalProto.TENSOR as in assert a few lines above, or is it supposed to the element type of tensor? I think the latter one. But I am unclear how a test-case would pass that assert above.

Reply:
actually looking at the caller the elem_type is OptionalProto.Tensor
if ele_type == onnx.mapping.OptionalProto.TENSOR:
            return onnx.helper.make_optional_tensor_value_info(
                name=name,
                elem_type=ele_type,
                shape=None)

what this means is this type will not work for make_tensor_value_info